nemo_text_processing/text_normalization/rw/taggers/cardinal.py

```diff
@@ -0,0 +1,319 @@
+# Copyright (c) 2024, NVIDIA CORPORATION & AFFILIATES.  All rights reserved.
+# Copyright (c) 2024, DIGITAL UMUGANDA
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import pynini
+from pynini.lib import pynutil
+import string
+from nemo_text_processing.text_normalization.en.graph_utils import GraphFst,NEMO_CHAR,insert_space
+from nemo_text_processing.text_normalization.rw.utils import get_abs_path
+
+def apply_fst(text, fst):
+    try:
+        print(pynini.shortestpath(text @ fst).string())
+        print(len(pynini.shortestpath(text @ fst).string()))
+
+    except pynini.FstOpError:
+        print(f"Error: no valid output with given'input: '{text}'")
+
+class CardinalFst(GraphFst):
+    def __init__(self):
+        super().__init__(name="cardinal", kind="classify")
+        alphabet = string.ascii_letters
+        rewrite_na_fst = pynini.cdrewrite(pynini.cross(" "," na "),pynini.union(*"aeiouAEIOU "),pynini.union(*"BCDFGHJKLMNPQRSTVWXYZbcdfghjklmnpqrstvwxyz"),NEMO_CHAR.closure())
```

Comment:
For line legitibility, mind just defining two string classes called `vowels` and `consonants` and inherit from your graph_fst class?